File: app/src/main/java/com/example/luisenriquez/mockdatavasev2/LoginActivity.java
package com.example.luisenriquez.mockdatavasev2;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class LoginActivity extends AppCompatActivity
{
    String usrNm, psswrd;
    EditText etUsrNm, etPsswrd;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        BindControls();

        instantiateUsers();
    }

    private void BindControls()
    {
        etUsrNm = findViewById(R.id.etEmail);
        etPsswrd = findViewById(R.id.etPassword);
    }

    private void instantiateUsers ()
    {
        usrNm = "MeMyself@gmail.com";
        psswrd = "password";
    }

    public void SignInOrRegister(View view)
    {
        validLoginInfo();
    }

    private void validLoginInfo()
    {
        String inUser = etUsrNm.getText().toString();
        String inPsswrd = etPsswrd.getText().toString();

        if (inUser.isEmpty())
        {
            etUsrNm.setError("Please fill user!");
            return;
        }

        if (inUser.equals(usrNm) && inPsswrd.equals(psswrd))
        {
            startActivity(new Intent(this, MainActivity.class));
        }
        else if (inUser.equals(usrNm) && !inPsswrd.equals(psswrd))
        {
            etPsswrd.setError("Incorrect Password");
        }
        else
        {
            startActivity(new Intent(this, RegisterActivity.class));
        }
    }
}

File: app/src/main/java/com/example/luisenriquez/mockdatavasev2/ui/activities/LoginActivity.kt
package com.example.luisenriquez.mockdatavasev2

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText

class LoginActivity : AppCompatActivity() {
    internal var usrNm: String
    internal var psswrd: String
    internal var etUsrNm: EditText
    internal var etPsswrd: EditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        BindControls()

        instantiateUsers()
    }

    private fun BindControls() {
        etUsrNm = findViewById(R.id.etEmail)
        etPsswrd = findViewById(R.id.etPassword)
    }

    private fun instantiateUsers() {
        usrNm = "MeMyself@gmail.com"
        psswrd = "password"
    }

    fun SignInOrRegister(view: View) {
        validLoginInfo()
    }

    private fun validLoginInfo() {
        val inUser = etUsrNm.text.toString()
        val inPsswrd = etPsswrd.text.toString()

        if (inUser.isEmpty()) {
            etUsrNm.error = "Please fill user!"
            return
        }

        if (inUser == usrNm && inPsswrd == psswrd) {
            startActivity(Intent(this, MainActivity::class.java))
        } else if (inUser == usrNm && inPsswrd != psswrd) {
            etPsswrd.error = "Incorrect Password"
        } else {
            startActivity(Intent(this, RegisterActivity::class.java))
        }
    }
}
